File: Medium/Java/209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int minLen = n + 1;
        int j = 0;
        int sum = 0;
        
        for(int i = 0; i < n; i++){
            sum += nums[i];
            while(target <= sum){
                int currLen = i - j + 1;
                minLen = Math.min(minLen, currLen);
                sum -= nums[j];
                j++;
            }
        }

        if(minLen != n + 1)
            return minLen;
        else
            return 0;
    }
}

File: Medium/Python/209. Minimum Size Subarray Sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        n = len(nums)
        minLen = n + 1
        j = 0
        this_sum = 0

        for i in range(n):
            this_sum += nums[i]
            while target <= this_sum:
                currLen = i - j + 1
                minLen = min(minLen, currLen)
                this_sum -= nums[j]
                j += 1

        if minLen != n + 1:
            return minLen
        else:
            return 0
